### PR TITLE
#59 - Improved types in AsyncResponse & AsyncSlice. Added tests

### LINKED ISSUE #59: AsyncSlice.java:41-44: Add unit tests for AsyncSlice and...
The puzzle `41-f448a190` from #41 has to be resolved: 

https://github.com/artipie/http/blob/612eb5835c4b3ce8b80061532741cba5f0aa308e/src/main/java/com/artipie/http/async/AsyncSlice.java#L41-L44

The puzzle was created by @g4s8 on 19-Feb-20. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #59 
Added tests and important typing improvements that make these classes convenient to use in adapters.

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 Please assign me

**Comment by 0crat** [comment]
This pull request [#104](https://github.com/artipie/http/issues/104) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/2db3db14-cf2e-4936-8f68-797630326aa4); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by olegmoz** [comment]
@paulodamaso ping

File: src/test/java/com/artipie/http/async/AsyncResponseTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+/**
+ * Tests for {@link AsyncResponse}.
+ *
+ * @since 0.8
+ */
+class AsyncResponseTest {
+
+    @Test
+    void shouldSend() {
+        final CompletableFuture<RsWithStatus> future = CompletableFuture.completedFuture(
```
Comment (paulodamaso) [inline src/test/java/com/artipie/http/async/AsyncResponseTest.java:45]: @olegmoz You could inline `future` declaration into the assertion

File: src/test/java/com/artipie/http/async/AsyncResponseTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Test
+    void shouldPropagateFailure() {
+        final CompletableFuture<Response> future = new CompletableFuture<>();
+        future.completeExceptionally(new IllegalStateException());
+        final CompletionStage<Void> result = new AsyncResponse(future)
+            .send((status, headers, body) -> CompletableFuture.allOf());
+        MatcherAssert.assertThat(
+            result.toCompletableFuture().isCompletedExceptionally(),
```
Comment (paulodamaso) [inline src/test/java/com/artipie/http/async/AsyncResponseTest.java:61]: @olegmoz Here also, please inline all inside the assertion

File: src/test/java/com/artipie/http/async/AsyncSliceTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * @since 0.8
+ */
+class AsyncSliceTest {
+
+    @Test
+    void shouldRespond() {
+        final CompletableFuture<SliceSimple> future = CompletableFuture.completedFuture(
+            new SliceSimple(new RsWithStatus(RsStatus.OK))
+        );
+        MatcherAssert.assertThat(
```
Comment (paulodamaso) [inline src/test/java/com/artipie/http/async/AsyncSliceTest.java:51]: @olegmoz Ditto

File: src/test/java/com/artipie/http/async/AsyncSliceTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            new RsHasStatus(RsStatus.OK)
+        );
+    }
+
+    @Test
+    void shouldPropagateFailure() {
+        final CompletableFuture<Slice> future = new CompletableFuture<>();
+        future.completeExceptionally(new IllegalStateException());
+        final CompletionStage<Void> result = new AsyncSlice(future)
+            .response("GET /index.html HTTP_1_1", Collections.emptySet(), Flowable.empty())
```
Comment (paulodamaso) [inline src/test/java/com/artipie/http/async/AsyncSliceTest.java:62]: @olegmoz here too

**Comment by olegmoz** [comment]
@paulodamaso please review the changes

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by paulodamaso** [comment]
@olegmoz There are still two comments to be fixed, please take a look

**Comment by olegmoz** [comment]
@paulodamaso what exactly has not been fixed? Everything is inlined in both classes

**Comment by paulodamaso** [comment]
@g4s8 Please merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**